bridge-impl/src/main/java/com/liferay/faces/bridge/component/internal/ComponentUtil.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-2016 Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package com.liferay.faces.bridge.component.internal;
+
+import javax.faces.component.UIComponent;
+import javax.faces.component.ValueHolder;
+
+/**
+ * @author  Kyle Stiemann
+ */
+public final class ComponentUtil {
+
+	private ComponentUtil() {
+		throw new AssertionError();
+	}
+
+	public static String getComponentValue(UIComponent componentResource) {
```

Comment:
@ngriffin7a, this method is in `ResourceUtil`, but that doesn't really seem appropriate for it. I think we should duplicate this functionality and avoid making a `getComponentValue()` method part of the API. (This class is in an `internal` package.) When I change the API of `ResourceVerifier`, I will change `ResourceUtil` as well.